src/Security/Authentication/Negotiate/src/Internal/ReflectedNegotiateState.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Security.Authentication;
+using System.Security.Principal;
+
+namespace Microsoft.AspNetCore.Authentication.Negotiate
+{
+    internal class ReflectedNegotiateState : INegotiateState
+    {
+        private static readonly ConstructorInfo _constructor;
+        private static readonly MethodInfo _getOutgoingBlob;
+        private static readonly MethodInfo _isCompleted;
+        private static readonly MethodInfo _protocol;
+        private static readonly MethodInfo _getIdentity;
+        private static readonly MethodInfo _closeContext;
+
+        private readonly object _instance;
+
+        static ReflectedNegotiateState()
+        {
+            var ntAuthType = typeof(AuthenticationException).Assembly.GetType("System.Net.NTAuthentication");
+            _constructor = ntAuthType.GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance).First();
+            _getOutgoingBlob = ntAuthType.GetMethods(BindingFlags.NonPublic | BindingFlags.Instance).Where(info =>
+                info.Name.Equals("GetOutgoingBlob") && info.GetParameters().Count() == 2).Single();
```

Comment:
Do you want to narrow down to NonPublic? If the names stay the same when the API becomes public, this will now break.

Reply:
The API proposal is for a new type with reduced surface area. See my branch here:
https://github.com/dotnet/corefx/blob/3e1bbb6cc6c201464f8371ba51c4ab9933e4d86c/src/System.Net.Security/src/System/Net/Security/NegotiateAuthState.cs

When that gets merged I will either have to be careful not to break myself though any private API  changes or react quickly to the breaks (which shouldn't be hard if the changes are constrained to this file).